src/core/unit.c

```diff
@@ -3380,7 +3380,10 @@ int unit_write_drop_in(Unit *u, UnitSetPropertiesMode mode, const char *name, co
         if (!dir)
                 return -EINVAL;
 
-        r = write_drop_in(dir, u->id, 50, name, data);
+        prefixed = strjoina("# This is a drop-in unit file extension, created via \"systemctl set-property\" or an equivalent operation. Do not edit.\n",
```

Comment:
I think the motivation is that another call to "set-property" will indiscriminately overwrite the file. It can be edited, but you risk losing your changes. Maybe a better text would be:
```
# This is a drop-in unit file extension, created via \"systemctl set-property\" or an equivalent operation.
# Changes will be overwritten when \"set-property\" is called again.
```